packages/mobile/src/screens/buy-sell-screen/BuySellFlow.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState, useEffect, useMemo, useCallback } from 'react'
+
+import { buySellMessages as messages } from '@audius/common/messages'
+import { Name } from '@audius/common/models'
+import type { BuySellTab } from '@audius/common/store'
+import {
+  useBuySellScreen,
+  useBuySellSwap,
+  useBuySellTabs,
+  useBuySellTransactionData,
+  useSwapDisplayData,
+  SUPPORTED_TOKEN_PAIRS,
+  useAddCashModal
+} from '@audius/common/store'
+import { useFocusEffect } from '@react-navigation/native'
+
+import { Button, Flex, Hint, TextLink } from '@audius/harmony-native'
+import { SegmentedControl } from 'app/components/core'
+import { useNavigation } from 'app/hooks/useNavigation'
+import { make, track } from 'app/services/analytics'
+
+import { BuyScreen, SellScreen } from './components'
+
+type BuySellFlowProps = {
+  onClose: () => void
+  initialTab?: 'buy' | 'sell'
+}
+
+const WALLET_GUIDE_URL = 'https://help.audius.co/product/wallet-guide'
+
+export const BuySellFlow = ({
+  onClose,
+  initialTab = 'buy'
+}: BuySellFlowProps) => {
+  const navigation = useNavigation()
+  const { onOpen: openAddCashModal } = useAddCashModal()
+
+  const { currentScreen, setCurrentScreen } = useBuySellScreen({
+    onScreenChange: () => {} // No-op since we handle navigation separately
+  })
+
+  const {
+    transactionData,
+    hasSufficientBalance,
+    handleTransactionDataChange,
+    resetTransactionData
+  } = useBuySellTransactionData()
+
+  const { activeTab, handleActiveTabChange } = useBuySellTabs({
+    setCurrentScreen,
+    resetTransactionData,
+    initialTab: (initialTab as BuySellTab) || 'buy'
```

Comment:
initialTab = 'buy' in params no not needed here right? and prob dont need to cast if we make buy-sell-flow props initialTab to be BuySellTab?